### PR TITLE
coord: remove unused byo timestamping formats

### PR DESCRIPTION
**Pull request by ruchirK**
We remove the ability for the timestamper thread to ingest consistency information
from anything that is not a Avro-formatted Debezium style consistency topic or AVRO
OCF. The thinking here is that our custom consistency formats (either byte/text
encoded or Avro encoded) should not be used in production, and users should prefer
either Debezium style consistency or CDCv2.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6142)

### CODE REVIEW

File: src/sql/src/plan/statement/ddl.rs [hunk 1]
```diff
@@ -593,6 +593,13 @@ pub fn plan_create_source(
                 cached_files: None,
             });
             let encoding = get_encoding(format)?;
+
+            if consistency != Consistency::RealTime
```
Comment (ruchirK) [inline src/sql/src/plan/statement/ddl.rs:597]: could someone from the sql team confirm if this kind of check is ok / if there is a better way to get what im going for here?

**Comment by ruchirK** [comment]
ooof so this is failing the new kafka exactly-once-sinks tests that use the old consistency formats - I'll port those over asap!

**Comment by ruchirK** [comment]
fixed the kafka eos test so this should be ready for a review!

**Review by elindsey — APPROVED** [review]
This lgtm! Seeing that many red lines in timestamp.rs is heartening. 🙂 

I'll defer to someone with more experience for your question about that planning check

**Review by frankmcsherry — APPROVED** [review]
I'm not morally opposed to this, but I am a bit weirded out that I do not know on what basis we might determine that it is ok to shed these implementations. If we had e.g. some design doc or guiding principles on possible source formats, possible sink formats, idk this sort of thing, then it would be more of a no-brainer to say "checks out; we don't need that stuff!".

But, as I have no personal attachment to these things I'm ok shedding them!

File: src/sql/src/plan/statement/ddl.rs (hunk 1 above)
Comment (sploiselle) [reply src/sql/src/plan/statement/ddl.rs:597]: I haven't touched the real time consistency code path at all, but given the prevalence of nearly identical checks in this function, this looks like the right approach. If there's a deeper apprehension, lmk and I can do a deeper dive.

Comment (ruchirK) [reply src/sql/src/plan/statement/ddl.rs:597]: thanks sean!

**Pull request merged**